### PR TITLE
Replace swiftlint with swiftformat

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
This PR replaces `swiftlint` with `swiftformat`. Although both spot and report style issues, swiftformat seems better at auto-formatting the code.

### Solution 📦
- Update `Rakefile` tasks.
- Add Danger plugin for Swiftformat.
- Remove `.swiftlint.yml`.
- Remove Swiftlint danger plugin.

@ollieatkinson is there any style configuration that you think we should include?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
 - Rename manifest group to `Manifest` https://github.com/tuist/tuist/pull/227 by @pepibumur.
 - Rename manifest target to `Project-Manifest` https://github.com/tuist/tuist/pull/227 by @pepibumur.
+- Replace swiftlint with swifromat https://github.com/tuist/tuist/pull/239 by @pepibumur.
```
Comment (ollieatkinson) [inline CHANGELOG.md]: s/swifromat/swiftformat

**Comment by ollieatkinson** [comment]
> @ollieatkinson is there any style configuration that you think we should include?

We should commit a .swiftformat file to ensure style is consistent across versions of the tool.
```
# file options

--symlinks ignore
--exclude Tests/XCTestManifests.swift

# format options

--allman false
--binarygrouping 4,8
--commas always
--comments indent
--decimalgrouping 3,6
--elseposition same-line
--empty void
--exponentcase lowercase
--exponentgrouping disabled
--fractiongrouping disabled
--header ignore
--hexgrouping 4,8
--hexliteralcase uppercase
--ifdef indent
--indent 4
--indentcase false
--importgrouping testable-bottom
--linebreaks lf
--octalgrouping 4,8
--operatorfunc spaced
--patternlet hoist
--ranges spaced
--self remove
--semicolons inline
--stripunusedargs always
--trimwhitespace always
--wraparguments preserve
--wrapcollections preserve
```

Taken from https://github.com/nicklockwood/SwiftFormat/blob/master/.swiftformat

**Comment by pepibumur** [comment]
Added @ollieatkinson

**Comment by marciniwanicki** [comment]
I might be wrong, the truth is that I don't SwiftFormat for that long and might miss some of its features, but I thought both swiftlint and SwiftFormat can play nicely together. There is a set of functionality that is overlapping but overall I think they have different purposes. SwiftFormat can go beyond formatting but it's still focused on making the code look "right" and consistent. On the other hand, swiftlint (even not as good as similar tools available for JVM languages) can catch cyclomatic complexity, large parameter lists, missing / redundant super calls and other code smells. I love nice-looking code but if I had to choose one, I think swiftlint would give us more value.

**Comment by marciniwanicki** [comment]
Please don't treat my previous comment as a blocker. I'd be happy to see this PR landed 👍. Maybe at some point we will reconsider adding swiftlint back (it wouldn't require significant code changes).

**Comment by pepibumur** [comment]
> I might be wrong, the truth is that I don't SwiftFormat for that long and might miss some of its features, but I thought both swiftlint and SwiftFormat can play nicely together. There is a set of functionality that is overlapping but overall I think they have different purposes. SwiftFormat can go beyond formatting but it's still focused on making the code look "right" and consistent. On the other hand, swiftlint (even not as good as similar tools available for JVM languages) can catch cyclomatic complexity, large parameter lists, missing / redundant super calls and other code smells. I love nice-looking code but if I had to choose one, I think swiftlint would give us more value.

Interesting point. I didn't know we could use swiftlint to catch complexity in the code. What about enabling Swiftlint for that kind of checks, and letting Swiftformat focus on the code style?

> Please don't treat my previous comment as a blocker. I'd be happy to see this PR landed 👍. Maybe at some point we will reconsider adding swiftlint back (it wouldn't require significant code changes).

No worries, we don't need to rush this one. I'm curious to know what @tuist/core thinks about having the two tools setup.
Personally, I think it's fine if we set up in such a way that they don't conflict.

**Comment by pepibumur** [comment]
@marciniwanicki I'm merging this one. We can re-evaluate the option if we see that it doesn't fit our needs.

**Pull request merged**